src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java

```diff
@@ -68,6 +71,38 @@ public CodeQualityServiceImpl(CodeQualityRepository codeQualityRepository,
         return searchType(request);
     }
 
+    @Override
+    public Iterable<CodeQuality> getAllSecurityScans(CodeQualityRequest request) {
+
+        Component component = componentRepository.findOne(request.getComponentId());
+
+        Iterable<CollectorItem> collectorItems = component.getCollectorItems(request.getType().collectorType());
+
+        ArrayList<CodeQuality> codeQualities = new ArrayList<CodeQuality>();
+
+        collectorItems.forEach(item -> {
+            CodeQuality tempCQ = codeQualityRepository.findTop1ByCollectorItemIdOrderByTimestampDesc(item.getId());
+
+            if(tempCQ != null){
```

Comment:
could be restructured, lets avoid huge if blocks

Reply:
Restructured